tests/test_type_alias_type.py

```diff
@@ -388,3 +389,56 @@ class Div(BaseModel):
             'type': 'literal_error',
         }
     ]
+
+
+def test_double_type_aliases():
+    # https://github.com/pydantic/pydantic/issues/8984
+    MySeq = TypeAliasType('MySeq', Sequence[T], type_params=(T,))
+    MyIntSeq = TypeAliasType('MyIntSeq', MySeq[int])
+
+    class MyModel(BaseModel):
+        my_int_seq: MyIntSeq
+
+    assert MyModel(my_int_seq=range(1, 4)).my_int_seq == [1, 2, 3]
+
+    assert MyModel.model_json_schema() == {
+        '$defs': {'MySeq_int_': {'items': {'type': 'integer'}, 'type': 'array'}},
+        'properties': {'my_int_seq': {'$ref': '#/$defs/MySeq_int_'}},
+        'required': ['my_int_seq'],
+        'title': 'MyModel',
+        'type': 'object',
+    }
+
+
+REQUIRE_PEP_695 = pytest.mark.skipif(sys.version_info < (3, 12), reason='requires Python 3.12+')
+
+
+@REQUIRE_PEP_695
+def test_double_type_aliases_pep695(create_module):
```

Comment:
Let's remove this one, it is functionally equivalent to the previous one.

Reply:
So is it a certain thing that `type A = ...` is merely a syntax sugar for `A = TypeAliasType('A', ...)`? If this is true then surely we can remove the test. I am just not sure because in the case of `def f[T](): ...`, `class A[T]: ...`, those are more than syntax sugars.

Reply:
> So is it a certain thing that `type A = ...` is merely a syntax sugar for `A = TypeAliasType('A', ...)`?

Yes, apart from the fact that a type alias type's `__value__` is lazily evaluated when using the `type` syntax, but the lazy evaluation is not relevant for our test.